### ISSUE TITLE
Should we fix the SpotBugs violation?

### ISSUE THREAD

**Issue by dengliming** [open-telemetry/opentelemetry-java-instrumentation#1166]
```
MS: io.opentelemetry.instrumentation.auto.api.concurrent.State.FACTORY isn't final but should be  At State.java:[line 29]
SpotBugs reported failures
org.gradle.api.GradleException: Verification failed: SpotBugs violation found: 1
	at com.github.spotbugs.snom.internal.SpotBugsRunnerForWorker$SpotBugsExecutor.execute(SpotBugsRunnerForWorker.java:95)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:49)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:43)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:97)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:43)
	at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:49)
	at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:30)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:85)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:55)
	at org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:138)
	at org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:135)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by iNikem** [comment]
We definitely have to decide if we want to use spotbugs or not. If we do, then we have to fix warnings or tune our configuration. If not, just remove it.

**Comment by dengliming** [comment]
Same as `[ant:checkstyle] `  lots of warn logs. : )
```
[ant:checkstyle] [WARN] D:\workspace4\dlm_github\opentelemetry-java-instrumentation\instrumentation\apache-httpasyncclient-4.0\src\main\java\io\opentelemetry\instrumentation\auto\apachehttpasyncclient\ApacheHttpAsyncClientInstrumentation.java:95:5: 缺少 Javadoc �? [JavadocMethod]
[ant:checkstyle] [WARN] D:\workspace4\dlm_github\opentelemetry-java-instrumentation\instrumentation\apache-httpasyncclient-4.0\src\main\java\io\opentelemetry\instrumentation\auto\apachehttpasyncclient\ApacheHttpAsyncClientInstrumentation.java:111:5: 缺少 Javadoc �? [JavadocMethod]
[ant:checkstyle] [WARN] D:\workspace4\dlm_github\opentelemetry-java-instrumentation\instrumentation\apache-httpasyncclient-4.0\src\main\java\io\opentelemetry\instrumentation\auto\apachehttpasyncclient\ApacheHttpAsyncClientInstrumentation.java:184:5: 缺少 Javadoc �? [JavadocMethod]
[ant:checkstyle] [WARN] D:\workspace4\dlm_github\opentelemetry-java-instrumentation\instrumentation\apache-httpasyncclient-4.0\src\main\java\io\opentelemetry\instrumentation\auto\apachehttpasyncclient\ApacheHttpAsyncClientTracer.java:52: 本行字符�? 111个，�?多：100个�?? [LineLength]
[ant:checkstyle] [WARN] D:\workspace4\dlm_github\opentelemetry-java-instrumentation\instrumentation\apache-httpasyncclient-4.0\src\main\java\io\opentelemetry\instrumentation\auto\apachehttpasyncclient\ApacheHttpAsyncClientTracer.java:53: 本行字符�? 111个，�?多：100个�?? [LineLength]
```

**Comment by iNikem** [comment]
Closed by #1325
